futures-util/src/stream/futures_unordered/mod.rs

```diff
@@ -155,6 +157,15 @@ impl<T> FuturesUnordered<T> {
         }
     }
 
+    /// Returns an iterator that allows modifying each future in the set.
```

Comment:
Can you add an `Unpin` bound to the `iter_mut` method itself? It could be confusing to have `iter_mut` be callable but the result not be usable as an iterator.